RGPosadas/LitterRally
# TASK-2: Setup GraphQL

**Task Description**
This PR addresses #10.

**Note about GraphQL decorators and code coverage**
If you see ```/* istanbul ignore next */``` on any of the GraphQL classes, it is to ignore them from the code coverage since the decorators take in anonymous functions.  
We have tried different ways of fixing it, like [this](https://github.com/gstamac/ts-jest/blob/master/README.md#ignore-coverage-on-decorators). The creator of type-graphql even dismissed an issue about a fix. [LINK](https://github.com/MichalLytek/type-graphql/issues/514) 😥

**Testing Steps**
1. Make sure to resolve all your dependencies with `npm i`
1. Run the application using ```nx serve mull-api```.
1. If you see something like this in the logs: 
![image](https://user-images.githubusercontent.com/36024638/96154374-b6f47480-0edc-11eb-8fa1-5368f910e780.png) 
It means that the GraphQL module has been loaded.
1. Open your browser and go to ```localhost:3333/graphql``` to enter the GraphiQL playground.
1. Test some queries and mutations. You should have the database open to check if your changes occur for mutations.

Additional information can be obtained by pressing on the DOCS and SCHEMA tabs or by watching the knowledge handover on [Slack](https://bug-bytes.slack.com/archives/CS5P4NL5Q/p1602637335001700).


 Example queries:
```graphql
{
  user(id:13) {
    id
    name
    description
    password
  }
}
```
```graphql
{
  user(id:12) {
    id
    name
    description
    password
    friends {
      name
    }
  }
}
```

Example mutations:
```graphql
mutation {
  createUser(createUserInput: {
    name:"George"
    email:"ge@orge.ca"
    password:"password"
    age:"22"
  }) {
    id
  }
}
```
```graphql
mutation {
  updateUser(updateUserInput: {
    id: 14
    age: "23"
  }) {
    id
  }
}
```
